src/testing/jest/README.md

```diff
@@ -0,0 +1,210 @@
+    --- a/renovate.json5
+    +++ b/renovate.json5
+    @@ -98,6 +98,11 @@
+           matchPackageNames: ['@types/jest', 'jest'],
+           allowedVersions: '<=27'
+         },
+    +    {
+    +      matchFileNames: ["src/testing/jest/jest-28/package.json"],
+    +      matchPackageNames: ['@types/jest', 'jest'],
+    +      allowedVersions: '<=28'
+    +    },
+         {
+    
+    ```
+   Note how:
+   - `matchFileNames` maps to the new Jest directory's `package.json`
+   - `matchPackageNames` lists the packages that were upgraded earlier
+   - `allowedVersions` allows versions less than or equal to the version we're adding support for
+1. Add the new version of Jest to the Component Starter smoke tests:
+    ```diff
+   index 16560a659..a4bee5308 100644
+    --- a/.github/workflows/test-component-starter.yml
+    +++ b/.github/workflows/test-component-starter.yml
+    @@ -16,7 +16,7 @@ jobs:
+    strategy:
+    fail-fast: false
+    matrix:
+    -        jest: ['24', '25', '26', '27']
+    +        jest: ['24', '25', '26', '27', '28']
+             node: ['16', '18', '20']
+             os: ['ubuntu-latest', 'windows-latest']
+      runs-on: ${{ matrix.os }}
+   ```
+1. Rename the `JestFacade` implementation in the new directory's `jest-facade.ts` file.
+   The number in the class name should match the version of Jest we're adding support for.
+   It is recommended that you use your editor's renaming/refactoring utility to do this step:
+    ```diff
+    import type * as d from '@stencil/core/internal';
+    import { isString } from '@utils';
+    
+    -import { Jest27Stencil } from './jest-facade';
+    +import { Jest28Stencil } from './jest-facade';
+    
+    /**
+    * Helper function for retrieving legacy Jest options. These options have been provided as defaults to Stencil users
+      @@ -147,7 +147,7 @@ export function buildJestConfig(config: d.ValidatedConfig): string {
+      jestConfig.verbose = stencilConfigTesting.verbose;
+      }
+    
+    -  jestConfig.testRunner = new Jest27Stencil().getDefaultJestRunner();
+    +  jestConfig.testRunner = new Jest28Stencil().getDefaultJestRunner();
+    
+       return JSON.stringify(jestConfig);
+       }
+       diff --git a/src/testing/jest/jest-28/jest-facade.ts b/src/testing/jest/jest-28/jest-facade.ts
+       index 3c7aa949c..3dea9b42b 100644
+       --- a/src/testing/jest/jest-28/jest-facade.ts
+       +++ b/src/testing/jest/jest-28/jest-facade.ts
+       @@ -10,7 +10,7 @@ import { jestSetupTestFramework } from './jest-setup-test-framework';
+       /**
+        * `JestFacade` implementation for communicating between this directory's version of Jest and Stencil
+        */
+    -  export class Jest27Stencil implements JestFacade {
+    +  export class Jest28Stencil implements JestFacade {
+        getJestCliRunner() {
+        return runJest;
+      }
+    
+    ```
+1. Update the `stencil-jest-connector.ts` to call the new `JestFacade` implementation:
```

Comment:
Unless this gets renamed in one of your other branches, I believe this should be `jest-stencil-connector.ts`

Reply:
D'Oh - https://github.com/ionic-team/stencil/pull/4952/commits/948af85113167a95b889c85eeb09bc2708e708cf